services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -880,6 +881,11 @@ def get_urlpaths_in_gen_kwargs(gen_kwargs: dict[str, Any]) -> list[str]:
     for shard in shards:
         if isinstance(shard, str):
             urlpaths.add(shard.split("::")[-1])
+        if shard and isinstance(shard, tuple):
+            if isinstance(shard[-1], FilesIterable):
```

Comment:
maybe a comment at least (or a test, at most) to give an example for each case

Reply:
I added comments as well as a test for every dataset builder and included normal data and zipped data